src/types.rs

```diff
@@ -30,6 +30,11 @@ macro_rules! iden_trait {
 
             fn unquoted(&self, s: &mut dyn fmt::Write);
         }
+
+        /// Identifier
+        pub trait IdenStatic: Iden  {
+            fn as_str(&self) -> &str;
+        }
```

Comment:
Can we make this definition same as SeaORM's ?

```rust
pub trait IdenStatic: Iden + Copy + Debug + 'static {
    /// Method to call to get the static string identity
    fn as_str(&self) -> &str;
}
```

This way the next step is to just remove the SeaORM version in the next major release.

Reply:
Can we implement `AsRef<str>`? And potentially an `as_str` that returns a `&'static str`

Reply:
@nitnelave Do you mean something like this:

```
impl<T> AsStr<str> for T 
where T: IdenStatic 
{
    ...
}
```
?

Reply:
Yes, or at least make it part of the derive (but a blanket implementation is probably better).

Reply:
Well, I try do it.

Reply:
@nitnelave I cannot do this for generic type, because I got error: `Only traits defined in the current crate can be implemented for arbitrary types [E0117]`

Reply:
@tyt2y3 or @billy1624 can someone explain why we need: `pub trait IdenStatic: Iden + Copy + Debug + 'static`? I understand only: `Iden` trait. other two - I don't...

Reply:
@nitnelave I add `AsRef<str>` implementation, can you review?

Reply:
Great question! I see your concern.

If we have `trait IdenStatic: Copy {}`, that means below isn't possible anymore because `String` doesn't `Copy`.

```rs
#[derive(Iden, EnumIter)]
enum Custom {
    Custom(String),
}
```

Reply:
I think `Debug` is just lazy-ness. It should be impl by hand. But we need that to be able to use `assert!`

Reply:
@billy1624 I think it's intentional that you can't derive `IdenStatic` for a `String`, no? Unless I'm mistaken, the idea behind `IdenStatic` is that it's just a straightforward compile-time mapping between the enum name and a static string. What would be the meaning of `Custom(String)` here?

Reply:
@nitnelave I think @billy1624 mean: https://github.com/SeaQL/sea-query/pull/508/files#diff-2a80134bd72b33f35551f1f2467f60acab30cad611bfc1c4999f59adeeecd221R20

Reply:
(I think your link got broken or something, it just links to the entire diff)

Reply:
Ups, I mean: https://github.com/SeaQL/sea-query/blob/master/sea-query-derive/tests/pass/renaming_everything.rs#L20

Reply:
I see. For me, that's a chief example of something that can't derive `IdenStatic`: it's not _actually_ static :)

Reply:
Yes you are right and I think it was what Billy meant as well.

Reply:
@tyt2y3 @nitnelave @billy1624 done. Now trait looks:

```
pub trait IdenStatic: Iden + Copy {
    fn as_str(&self) -> &'static str;
}
```